core/src/main/java/io/github/tcdl/Consumer.java

```diff
@@ -31,7 +30,7 @@
     private ChannelMonitorAgent channelMonitorAgent;
     private Clock clock;
 
-    private List<Subscriber> subscribers;
+    private ConcurrentLinkedQueue<Subscriber> subscribers;
```

Comment:
Do we really need to have concurrent collection here? Seems like methods that modify it are called from other synchronized methods. This double synchronization seems to be an overkill.